src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,115 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int NOT_EXISTS = -1;
+    private static final int UNIT = 1;
+    private T[] internalArray;
+    private int capacity;
+    private int size;
+
+    public ArrayList() {
+        capacity = 10;
+        internalArray = (T[]) new Object[capacity];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size < capacity) {
+            internalArray[size] = value;
+            size++;
+        } else {
+            growIfArrayFull();
+            add(value);
+        }
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (size == index) {
+            add(value);
+        } else if (isIndexAcceptable(index)) {
+            size++;
+            if (size == capacity) {
+                growIfArrayFull();
+            }
+            System.arraycopy(internalArray, index, internalArray,
+                    index + UNIT, size - index - UNIT);
```

Comment:
From `checklist.md`:
> Resize the array in a separate method.

Reply:
![image](https://github.com/mate-academy/jv-array-list/assets/90136583/ec09e8ba-5484-4449-a296-2e5debc6227e)
Is a growIfArrayFull() method is not a separate method for resizing?